cmake/install_png.cmake

```diff
@@ -0,0 +1,135 @@
+include(ExternalProject)
+if(NOT ${PENGUINV_USE_EXTERNAL_PNG})
```

Comment:
My idea for this file is:
```bash
include(ExternalProject)

option(PENGUINV_ENABLE_PNG_SUPPORT "Enable support of libpng" ON)
option(PENGUINV_USE_EXTERNAL_PNG "Download libpng and build from source" OFF)

if(NOT ${PENGUINV_ENABLE_PNG_SUPPORT})
    find_package(PNG REQUIRED)
    if(NOT PNG_FOUND)
        set(PENGUINV_USE_EXTERNAL_PNG "ON")
    endif()
endif()

if(${PENGUINV_ENABLE_PNG_SUPPORT})
    current code for external support
```

Explanation:
1. We declare variable `PENGUINV_ENABLE_PNG_SUPPORT` only when we include this file, intentionally knowing this. Currently if I want to run cmake file only for unit tests, it would fail due to unset variables.
2. By default we try to find png package. If we failed to find then we would try to use external source of installation.
3. We remove variable declarations from main Cmake file as we declare them here.
4. We remove variable setup in `appveyor.yml` file as well.

Of course we might need to move
```bash
if(NOT CMAKE_MODULE_PATH)
    set(CMAKE_MODULE_PATH)
endif()
list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
```
logic in cmake for unit tests. This is I'm not sure about right now.

Reply:
Done in 07d979f